essencium-backend/src/main/java/de/frachtwerk/essencium/backend/security/oauth2/OAuth2SuccessHandler.java

```diff
@@ -90,67 +90,75 @@ public void onAuthenticationSuccess(
       redirectHandler.setDefaultTargetUrl(oAuth2ConfigProperties.getDefaultRedirectUrl());
     }
 
-    if (!(authentication instanceof OAuth2AuthenticationToken)) {
-      LOGGER.error(
-          "did not receive an instance of {}, aborting",
-          OAuth2AuthenticationToken.class.getSimpleName());
-      redirectHandler.onAuthenticationSuccess(request, response, authentication);
-      return;
-    }
-
-    final var providerName =
-        ((OAuth2AuthenticationToken) authentication).getAuthorizedClientRegistrationId();
-
-    UserInfoEssentials userInfo;
-    try {
-      LOGGER.info(
-          "attempting to log in oauth2 user '{}' using provider '{}'",
-          authentication.getName(),
-          providerName);
-      userInfo = extractUserInfo((OAuth2AuthenticationToken) authentication, providerName);
-    } catch (UserEssentialsException e) {
-      LOGGER.error(e.getMessage());
-      redirectHandler.onAuthenticationSuccess(request, response, authentication);
-      return;
-    }
-
-    try {
-      final var user = userService.loadUserByUsername(userInfo.getUsername());
-      LOGGER.info("got successful oauth login for {}", userInfo.getUsername());
-
-      HashMap<String, Object> patch = new HashMap<>();
-
-      patch.put("firstName", userInfo.getFirstName());
-      patch.put("lastName", userInfo.getLastName());
-
-      if (oAuth2ConfigProperties.isUpdateRole()) {
-        List<Role> roles =
-            extractUserRole(((OAuth2AuthenticationToken) authentication).getPrincipal());
-        Role defaultRole = roleService.getDefaultRole();
-        if (roles.isEmpty() && Objects.nonNull(defaultRole)) {
-          LOGGER.info("no roles found for user '{}'. Using default Role.", userInfo.getUsername());
-          roles.add(defaultRole);
-        }
-        patch.put("roles", roles);
+    if (authentication instanceof OAuth2AuthenticationToken oAuth2AuthenticationToken) {
+      final String providerName = oAuth2AuthenticationToken.getAuthorizedClientRegistrationId();
+
+      final OAuth2ClientRegistrationProperties.ClientProvider clientProvider =
```

Comment:
Can the result of `oAuth2ClientRegistrationProperties.getProvider().get(providerName);` be `null`? From my point of view, if the result is`null` an Exception should be thrown right after the call